python/tank/bootstrap/configuration.py

```diff
@@ -78,18 +78,21 @@ def get_tk_instance(self, sg_user):
         path = self._path.current_os
         core_path = get_core_python_path_for_config(path)
 
+        # Get the user before the core swapping.
+        from ..authentication import serialize_user
```

Comment:
instead of passing functions around, would it be possible to simply 
- serialize the user into string
- swap the core 
- set the authenticated user for the new core and as part of this pass in the serialized user.

Reply:
we have a general problem with this kind of pattern - i wonder if we should handle it in a general way rather than one-off. would love to chat about it.

What if the site core supports feature X but the core you are switching to doesn't because it is old. What do we do? there should be a way to be able to detect that and report an error message early, before we perform the swapping i think.